x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -665,6 +669,58 @@ export class MapEmbeddable
     } as ActionExecutionContext;
   };
 
+  // remove legacy library tranform methods
+  linkToLibrary = undefined;
+  unlinkFromLibrary = undefined;
+  // add implemenation for library transform methods
+  saveStateToSavedObject = async (title: string) => {
+    const { attributes, references } = extractReferences({
+      attributes: this._savedMap.getAttributes(),
+    });
+
+    const {
+      item: { id: savedObjectId },
+    } = await getMapClient().create({
+      data: {
+        ...attributes,
+        title,
+      },
+      options: { references },
+    });
+    return {
+      state: {
+        ..._.omit(this.getExplicitInput(), 'attributes'),
+        savedObjectId,
+      },
+      savedObjectId,
+    };
+  };
+  savedObjectAttributesToState = () => {
```

Comment:
When is this method used again? Is this called during the unlink process?

Reply:
Correct, its used in src/plugins/dashboard/public/dashboard_actions/unlink_from_library_action.tsx

Should I add comments to HasLibraryTransform interface to show which methods are used for linking and which are used for unlinking?

Reply:
I think some extra documentation could help sure! The unlink method takes whatever you return there, and basically just replaces the panel with that state right?

Reply:
right. I will add some documentation

Reply:
Comments added in https://github.com/elastic/kibana/pull/178584/commits/298f8acc677cd3f8e4cc381ed81529efe33013d2. Let me know if more is needed or if these don't make sense

Reply:
Comments look great thanks!